openrtb_ext/request.go

```diff
@@ -308,3 +310,576 @@ type ExtRequestPrebidDataEidPermission struct {
 	Source  string   `json:"source"`
 	Bidders []string `json:"bidders"`
 }
+
+// RequestWrapper wraps the OpenRTB request to provide a storage location for unmarshalled ext fields, so they
+// will not need to be unmarshalled multiple times.
+type RequestWrapper struct {
+	// json json.RawMessage
+	Request *openrtb2.BidRequest
+	// Dirty bool // Probably don't care
+	UserExt    *UserExt
+	DeviceExt  *DeviceExt
+	RequestExt *RequestExt
+	AppExt     *AppExt
+	RegExt     *RegExt
+	SiteExt    *SiteExt
+}
+
+type UserExt struct {
+	Ext            map[string]json.RawMessage
+	Consent        *string
+	ConsentDirty   bool
+	Prebid         *ExtUserPrebid
+	PrebidDirty    bool
+	DigiTrust      *ExtUserDigiTrust
+	DigiTrustDirty bool
+	Eids           *[]ExtUserEid
+	EidsDirty      bool
+}
+
+func (ue *UserExt) Extract(extJson json.RawMessage) (*UserExt, error) {
```

Comment:
That may be a better way to go, but then they will all need to be given unique names. The current way would be good if we want some interfaces to handle the extract and marshal functionality, but I am not sure the case is there to do it via a loop over interfaces rather than by explicit references. So basically I am open, do you think it is worth the bit of recoding necessary?